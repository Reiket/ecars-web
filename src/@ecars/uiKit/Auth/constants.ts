import {withBlockClass} from 'ecars-web-lib';
import {AuthPageWrapper} from '@ecars/uiKit/Auth/components/AuthPageWrapper';

export const AuthPageWrapperHOC = withBlockClass(AuthPageWrapper, 'auth');