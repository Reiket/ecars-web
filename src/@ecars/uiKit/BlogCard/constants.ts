import type {CategoriesListType} from 'ecars-web-lib';

export const BLOG_CARD_DIRECTION = Object.freeze({
  ROW: 'row',
  COLUMN: 'column',
});

export type BlogCardDirectionType = (typeof BLOG_CARD_DIRECTION)[keyof typeof BLOG_CARD_DIRECTION];

export interface BlogCardContentProps {
  title: string;
  category: CategoriesListType;
  description?: string;
}

export interface BlogCardImageProps {
  imageUrl: string;
  alt?: string;
}

export const BLOG_CARD_TEST_ID = 'blogCardTestId';
