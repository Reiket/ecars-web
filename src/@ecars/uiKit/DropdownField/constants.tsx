import type {DropdownOption} from 'ecars-web-lib';

interface TopbarDropdownsConfig {
  label: string;
  options: DropdownOption[];
}

const shipToConfig: DropdownOption[] = [
  {
    value: 'ca',
    name: 'Canada',
    label: (
      <img
        src="./assets/images/flags/canada.svg"
        alt="ca"
      />
    ),
  },
  {
    value: 'us',
    name: 'USA',
    label: (
      <img
        src="./assets/images/flags/usa.svg"
        alt="usa"
      />
    ),
  },
  {
    value: 'eu',
    name: 'Europe',
    label: (
      <img
        src="./assets/images/flags/europe.svg"
        alt="eu"
      />
    ),
  },
];

const currencyConfig: DropdownOption[] = [
  {
    value: 'usd',
    name: 'US Dollar',
  },
  {
    value: 'eur',
    name: 'Euro',
  },
  {
    value: 'gbp',
    name: 'British Pound',
  },
];

export const topbarDropdownsConfig: TopbarDropdownsConfig[] = [
  {
    label: 'Ship to:',
    options: shipToConfig,
  },
  {
    label: 'Currency:',
    options: currencyConfig,
  },
];
