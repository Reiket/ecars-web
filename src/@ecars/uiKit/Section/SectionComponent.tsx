import type {FC, ReactNode} from 'react';
import {Section} from '@ecars/uiKit/Section/index';
import type {SectionContainerSize} from '@ecars/uiKit/Section/constants';
import type {ElementProps} from 'ecars-web-lib';
import {cn} from 'ecars-web-lib';

export interface Props extends ElementProps {
  children: ReactNode;
  containerSize?: SectionContainerSize;
}

export const SectionComponent: FC<Props> = ({block, children, containerSize}) => (
  <section className={cn(block, 'section')}>
    <Section.Container
      block={block}
      size={containerSize}
    >
      {children}
    </Section.Container>
  </section>
);
