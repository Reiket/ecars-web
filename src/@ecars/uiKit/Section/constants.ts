export const SECTION_CONTAINER_SIZE = Object.freeze({
  SM: 'sm',
  MD: 'md',
  LG: 'lg',
});

export type SectionContainerSize = (typeof SECTION_CONTAINER_SIZE)[keyof typeof SECTION_CONTAINER_SIZE];
