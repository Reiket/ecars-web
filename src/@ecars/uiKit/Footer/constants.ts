import {withBlockClass} from 'ecars-web-lib';
import {FooterWrapper} from '@ecars/uiKit/Footer/components/FooterWrapper';
import {socialLinks} from '@ecars/uiKit/SocialList/constants';
import {PageUrls} from '@ecars/constants/page-urls';
import {FooterCopyWrapper} from '@ecars/uiKit/Footer/components/FooterCopyWrapper';

export const FooterWrapperHOC = withBlockClass(FooterWrapper, 'footer');
export const FooterCopyWrapperHOC = withBlockClass(FooterCopyWrapper, 'copy');
export const footerSocialLinks = socialLinks.filter((item) => !('label' in item));
export const footerContactsLinks = socialLinks.filter((item) => 'label' in item);

export const copyLinks = [
  {
    label: 'Privacy Policy',
    href: PageUrls.PRIVACY,
  },
  {
    label: 'Terms & Conditions',
    href: PageUrls.TERMS,
  },
];

export interface FooterLink {
  label: string;
  to: string;
}

export const companyLinks: FooterLink[] = [
  {label: 'About Us', to: PageUrls.ABOUT},
  {label: 'Blog', to: PageUrls.Blog},
  {label: 'FAQ', to: PageUrls.FAQ},
];

export const carsLinks: FooterLink[] = [
  {label: 'Special offers', to: '/'},
  {label: 'New cars', to: PageUrls.NEWCARS},
  {label: 'Used cars', to: PageUrls.USEDCARS},
  {label: 'Brands', to: '/'},
];
