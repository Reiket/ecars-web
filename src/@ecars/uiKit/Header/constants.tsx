import {Icons, withBlockClass} from 'ecars-web-lib';
import {HeaderWrapper} from '@ecars/uiKit/Header/components/HeaderWrapper';
import type {PageUrlType} from '@ecars/constants/page-urls';
import {PageUrls} from '@ecars/constants/page-urls';
import {HeaderNavbarWrapper} from '@ecars/uiKit/Header/components/HeaderNavbarWrapper';
import type {ReactNode} from 'react';

export const HeaderWrapperHOC = withBlockClass(HeaderWrapper, 'header');
export const HeaderNavbarWrapperHOC = withBlockClass(HeaderNavbarWrapper, 'navbar');

interface NavbarLink {
  to: PageUrlType;
  label: string;
  isUtil?: boolean;
}

interface HeaderUtil {
  to: PageUrlType;
  icon: ReactNode;
}

export const navbarLinksConfig: NavbarLink[] = [
  {
    to: PageUrls.CATALOG,
    label: 'All Cars',
  },
  {
    to: PageUrls.ABOUT,
    label: 'About Us',
  },
  {
    to: PageUrls.Blog,
    label: 'Blog',
  },
  {
    to: PageUrls.FAQ,
    label: 'FAQ',
  },
  {
    to: PageUrls.FAVORITES,
    label: 'Favorites',
    isUtil: true,
  },
  {
    to: PageUrls.LOGIN,
    label: 'Login',
    isUtil: true,
  },
];

export const headerUtilsConfig: HeaderUtil[] = [
  {to: PageUrls.FAVORITES, icon: <Icons.FavoriteSolid />},
  {to: PageUrls.LOGIN, icon: <Icons.Account />},
];

export const headerTestScenarios = [
  {
    desc: 'initial state',
    scrolls: [0],
    expected: {sticky: false, showHeader: true},
  },
  {
    desc: 'scrolled down below topbarHeight → sticky true, showHeader false',
    scrolls: [100],
    expected: {sticky: true, showHeader: false},
  },
  {
    desc: 'scroll up again → sticky true, showHeader true',
    scrolls: [150, 50],
    expected: {sticky: true, showHeader: true},
  },
];
