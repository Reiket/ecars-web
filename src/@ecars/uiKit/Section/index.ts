import {SectionComponent} from '@ecars/uiKit/Section/SectionComponent';
import {SectionContainer} from '@ecars/uiKit/Section/components/SectionContainer';

export const Section = Object.assign(SectionComponent, {
  Container: SectionContainer,
});
