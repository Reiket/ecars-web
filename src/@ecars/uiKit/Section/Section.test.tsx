import {SECTION_CONTAINER_SIZE} from '@ecars/uiKit/Section/constants';
import {Section} from '@ecars/uiKit/Section/index';
import {render} from '@testing-library/react';

describe('Section Component', () => {
  const containerSizes = Object.values(SECTION_CONTAINER_SIZE);
  containerSizes.forEach((size) => {
    test('render component correctly', () => {
      const {container} = render(
        <Section
          containerSize={size}
          block="test"
        >
          1
        </Section>,
      );
      expect(container).toMatchSnapshot();
      const containerElement = container.querySelector('.container');
      expect(container).toBeInTheDocument();
      expect(containerElement).toHaveClass(`container--${size}`);
    });
  });
});
