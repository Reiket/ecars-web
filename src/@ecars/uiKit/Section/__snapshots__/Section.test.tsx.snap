// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Section Component > render component correctly 1`] = `
<div>
  <section
    class="test__section section"
  >
    <div
      class="test__container container container--sm"
    >
      1
    </div>
  </section>
</div>
`;

exports[`Section Component > render component correctly 2`] = `
<div>
  <section
    class="test__section section"
  >
    <div
      class="test__container container container--md"
    >
      1
    </div>
  </section>
</div>
`;

exports[`Section Component > render component correctly 3`] = `
<div>
  <section
    class="test__section section"
  >
    <div
      class="test__container container container--lg"
    >
      1
    </div>
  </section>
</div>
`;
